db/migrations/001_create_users.cr

```diff
@@ -2,7 +2,11 @@ class CreateUsers::V001 < LuckyMigrator::Migration::V1
   def migrate
     create :users do
       add first_name : String, index: true
+      add last_name : String
     end
+
+    add_index :users, :last_name, unique: true
+    # add_index :users, :first_name, unique: true # Note: breaks the migration
```

Comment:
Why does this break the migration? Is it because one was already added? If so I think we can remove this and create a separate issue for showing a better issue if an index already exists

Reply:
Did you see my comment on #27 ? Essentially since add_index is called after create, the table is created but not tracked. So you can't rollback the migration (since it's not in the migrations table) or migrate again because the table exists.

Reply:
Ah yes I see. I think this comment can be removed. It should be fixed once transactions are added: https://github.com/luckyframework/migrator/issues/27#issuecomment-345294466